### PR TITLE
[REF] purchase_console: Define a new controller in order to handle the website actions and Allow editing cells values

### PR DESCRIPTION
**Pull request by hugho-ad**
DUMMY BUILD
https://github.com/Vauxoo/yoytec/pull/2001

Added double click event to table rows in order to allow edition cell values

image
- controllers/main.py:
  Define a new controller in order to handle the website actions:
    Procure: 'procure_products_from_suppliers',
    Send RFQ: 'sent_suppliers',
    Open Bid: 'open_bid',
    Generate PO: 'generate_po',
    Cancel all: 'cancel_requisition'
- static/css/console.css
  Added some styles in order to display a loading overlay while
  asynchronous request is done
- static/js/console_frontend.js
  Bind event listener for action buttons
- static/js/web_widget_one2many_console.js
  Remove unnecessary overwriting of init method
- static/xml/web_widget_one2many_console.xml
  Removed log statement.
- views/layout.xml
  Added missing elements to match the table rows in backend
- views/purchase_requisition_view.xml
  Define a new Form view for ´purchase.requisition.line´ model
  and remove the attribute editable='bottom'
  until the editable widget one2many_columns is done.
- Now the action buttons are dynamically hide
- Depending on the qty of partners in the header, all lines will be set to the same width, this will make that we save space on the layout.
- Migrate css stylish to less file in order to separate the front end and backend styles
  - Fix the width of the cells inputs when editing the value
  - Disable enter the interval 48 to 57 eventes in the input in order to allow only numbers and dot into cell values
- Refactory _get_po_line method to avoid search inside a for loop instead of make a filter in po_lines

### CODE REVIEW

**Comment by hugho-ad** [comment]
@nhomar 
can you review it please?

**Comment by hugho-ad** [comment]
@nhomar @moylop260 can you review please

File: purchase_console/models/requisition.py [hunk 1]
```diff
@@ -84,12 +84,33 @@ def _inverse_suppliers(self):
                                                " do you have of stock.")
 
     @api.multi
+    def confirm_call(self):
+        self.ensure_one()
```
Comment (moylop260) [inline purchase_console/models/requisition.py:88]: Is really need the `self.ensure_one()` here?

FYI we can use `self.write` for singleton and non-singleton records

**Review by moylop260 — APPROVED** [review]
LGTM just small comment
I'll merge it but consider change in other PR my feedback if you think is fine

File: purchase_console/models/requisition.py (hunk 1 above)
Comment (hugho-ad) [reply purchase_console/models/requisition.py:88]: My intention is process one record at time, so I consider that is fine.

**Pull request merged**